server/controllers/comments.js

```diff
@@ -22,20 +35,49 @@ const getCommentsOfPost = async (req, res) => {
   }
 };
 
+// @route   GET '/api/comments/:commentId/replies'
+// @desc    get all replies of a specific comment
+// @access  private
+const getRepliesOfAComment = async (req, res) => {
+  try {
+    const comment = await getComment(req.params.commentId);
+
+    if (!comment) {
+      return serverResponse(res, 404, { message: "Comment doesn't exist" });
+    }
+
+    const replies = await getRepliesOfComment(req.params.commentId);
+
+    return serverResponse(res, 200, replies);
+  } catch (e) {
+    return serverResponse(res, 500, {
+      message: 'Internal error while trying to get all replies of this comment'
+    });
+  }
+};
+
 // @route   POST '/api/posts/:postId/comments'
 // @desc    add a comment to a post
 // @access  private
 const addCommentToPost = async (req, res) => {
   try {
+    const post = await getPost(req.params.postId);
+
+    if (!post) {
+      return serverResponse(res, 404, { message: "Post doesn't exist" });
+    }
+
     const comment = {
       ...req.body,
       user: req.user.sub,
       post: req.params.postId
     };
-    const post = await getPost(req.params.postId);
+
     post.comments++;
     await post.save();
+
     const response = await addComment(comment);
```

Comment:
This can be made faster with Promise.all no reason for them to wait for one another